### DISCUSSION TITLE
Migration sur ES 8

### DISCUSSION THREAD

**Discussion by m-maillot** [SocialGouv/code-du-travail-numerique#5842] [Technique]
# Contexte

Le code du travail numérique utilise actuellement ES en version 7. Cette dernière arrive en fin de vie eti l faut planifier la migration sur ES 8.

# Impacts

Les impacts ne sont pas tous connus, il faut faire une première version pour identifier les impacts sur les différentes applications.

On identifie 3 projets qui seront impactés :

 * cdtn-admin : l'administration met à jour les documents sur ES
 * code-du-travail-numerique : le site utilise ES comme base de données. ES est également utilisée pour le moteur de recherche.
 * cdtn-monolog : les analytics du site sont construit à partir de données dans ES. 

# Etapes

Il va y avoir deux grandes étapes. La migration de l'ES pour le site et ensuite pour les analytics.

## Migration pour le site

Voici les tâches à réaliser :

Partie admin : 

- [x] Création d'un ES en version 8 (demande aux ops)
- [x] Création d'une branche linked sur l'admin pour migrer ce nouveau ES
- [x] Mise à jour des documents sur cet nouveau ES

Cette partie ne devrait pas rencontré de soucis, l'admin ne fait qu'insérer des données dans ES. Il n'y a pas de requêtage.

Partie site :

- [x] Création d'une branche linked qui utilise le nouveau ES
- [x] Migrer le code pour utiliser ES 8 (code qui compile et TU ok)
- [x] Tester les différentes pages du site
- [x] Tester le moteur de recherche (faire un snapshot des résultats des top requêtes sur la prod et comparer avec celles de la branche)
- [x] Tester l'outil rechercher sa CC

Cette partie peut poser des complications sur la partie recherche. Cette partie est très complexe et utilise des requêtes complexes. Le reste devrait se faire sans grands accrocs.

La dernière étape consistera à valider en preprod avant la mise en prod.

**Comment by maxgfr** [comment]
+1

**Comment by carolineBda** [comment]
Du coup on fera monolog dans un 2ème temps ?
